flow/project.py

```diff
@@ -1582,14 +1583,18 @@ def _generate_operations(self, cmd, jobs, requires=None):
 
     def _get_pending_operations(self, jobs, operation_names=None):
         "Get all pending operations for the given selection."
-        operation_names = None if operation_names is None else set(operation_names)
+        if operation_names is None:
+            names = patterns = None
+        else:
+            patterns = [name.lstrip('/') for name in operation_names if name.startswith('/')]
+            names = [name for name in operation_names if not name.startswith('/')]
 
-        for op in self.next_operations(*jobs):
-            if operation_names and op.name not in operation_names:
-                continue
-            if not self.eligible_for_submission(op):
```

Comment:
I don't think we want to lose the eligibility check here because the function is also used for determining operations to run and operations to generate scripts for (in addition to in submission).

Reply:
Yes, exactly. And so it makes more sense to check the "eligible_for_submission()" in the `submit` function, see here: https://github.com/glotzerlab/signac-flow/blob/e2d3cf18a51a52e3f86a157d8df2e401f1ace9ad/flow/project.py#L1819

Reply:
Yes, exactly. And so it makes more sense to check the "eligible_for_submission()" in the `submit()` function, see here: https://github.com/glotzerlab/signac-flow/blob/e2d3cf18a51a52e3f86a157d8df2e401f1ace9ad/flow/project.py#L1819

Reply:
I got confused reading the diff here and didn't realize that `next_operations` was being used, so I was concerned that we were never checking `op.eligible`. I wanted to make sure that we weren't losing that check while removing the `eligible_for_submission` check, so I agree that we're good.